cmd/execution/main.go

```diff
@@ -564,6 +605,55 @@ func main() {
 		}).Run()
 }
 
+// getContractEpochCounter Gets the epoch counters from the FlowEpoch smart contract from the latest executed block.
+func getContractEpochCounter(vm *fvm.VirtualMachine, vmCtx fvm.Context, executionState state.ExecutionState) (uint64, error) {
+	// Get the address of the FlowEpoch smart contract
+	sc, err := systemcontracts.SystemContractsForChain(vmCtx.Chain.ChainID())
+	if err != nil {
+		return 0, err
+	}
+	address := sc.Epoch.Address
+
+	// Generate the script to get the epoch counter from the FlowEpoch smart contract
+	scriptCode := templates.GenerateGetCurrentEpochCounterScript(templates.Environment{
+		EpochAddress: address.Hex(),
+	})
+	script := fvm.Script(scriptCode)
+
+	// Get latest executed block and a view at that block
+	ctx := context.Background()
+	_, blockID, err := executionState.GetHighestExecutedBlockID(ctx)
```

Comment:
should we compare the epoch counter for the same block? it's possible that we are reading epoch counter from smart contract for block A , but reading epoch counter from protocol state for block B, and I'm not sure comparing them is accurate.

cc @jordanschalm 

However, should we compare the same block for the last finalized block? no, because a finalized block might not be executed yet, in which case we don't have the execution state to read from.

Maybe we should read the epoch counter for the last executed block, which guarantees we can read from both protocol state and smart contract. 

Also we are reading epoch counter for the same block, we'd better log the block view and block ID, along with the epoch counter.

Reply:
> Maybe we should read the epoch counter for the last executed block, which guarantees we can read from both protocol state and smart contract.

Yes, great point. We should definitely do this.

Reply:
I changed it so they should use the same block now.